### PR TITLE
Service check for y2nfs

### PR DESCRIPTION
**Pull request by coolgw**
[Add y2nfs service check before/after migration]

- Related ticket: https://progress.opensuse.org/issues/55004
- Needles: n/a
- Verification run: 
latest run2109-8-19:
http://openqa.suse.de/tests/3264435 for service check on  migration scenario case
http://openqa.suse.de/tests/3264436 for function server check
run2019-08-20
http://openqa.suse.de/tests/3270133#
http://openqa.suse.de/tests/3270134#

### CODE REVIEW

File: lib/nfs_common.pm [hunk 1]
```diff
@@ -183,5 +183,85 @@ sub yast2_server_initial {
     } while (not match_has_tag('nfs-config'));
 }
 
+sub install_service {
+    # Make sure packages are installed
+    zypper_call 'in yast2-nfs-server nfs-kernel-server', timeout => 480, exitcode => [0, 106, 107];
```
Comment (Vogtinator) [inline lib/nfs_common.pm]: IMO only an exit code of 0 is acceptable.

Comment (Vogtinator) [inline lib/nfs_common.pm]: `yast2-nfs-server` should take care of installing `nfs-kernel-server`. Now this isn't tested anymore.

File: lib/nfs_common.pm [hunk 2]
```diff
@@ -183,5 +183,85 @@ sub yast2_server_initial {
...
+    add_shares($rw, $ro);
+
+    send_key 'alt-f';
+    wait_serial("$module_name-0") or die "'yast2 $module_name' didn't finish";
+
+}
+
+sub start_service {
+    my ($rw, $ro) = @_;
+    # Back on the console, test mount locally
```
Comment (Vogtinator) [inline lib/nfs_common.pm]: This function doesn't test mounting itself though, so this comment is now wrong

File: lib/nfs_common.pm [hunk 3]
```diff
@@ -183,5 +183,85 @@ sub yast2_server_initial {
...
+        validate_script_output "nfsstat -m", sub { m/vers=3/ };
+    }
+
+    assert_script_run "exportfs | grep '${rw}\\|${ro}'";
+    assert_script_run "cat /etc/exports | tr -d ' \\t\\r' | grep '${rw}\\*(rw,\\|${ro}\\*(ro,'";
+    assert_script_run "cat /proc/fs/nfsd/exports";
+
+    assert_script_run('systemctl is-enabled nfs-server');
+    assert_script_run('systemctl is-active nfs-server');
+
```
Comment (Vogtinator) [inline lib/nfs_common.pm]: Whitespace

File: lib/nfs_common.pm [hunk 4]
```diff
@@ -183,5 +183,85 @@ sub yast2_server_initial {
...
+    # Check NFS version
+    if ((is_sle('=12-sp1') || is_sle('=12-sp2')) && script_run 'nfsstat -m') {
+        record_soft_failure 'bsc#1140731';
+    }
+    else {
+        validate_script_output "nfsstat -m", sub { m/vers=3/ };
+    }
+
+    assert_script_run "exportfs | grep '${rw}\\|${ro}'";
```
Comment (Vogtinator) [inline lib/nfs_common.pm]: That would succeed even if only one of them is actually there

File: lib/nfs_common.pm [hunk 5]
```diff
@@ -183,5 +183,85 @@ sub yast2_server_initial {
...
+
+}
+
+sub check_y2_nfs_func {
+    my ($stage) = @_;
+    my $rw      = '/srv/nfs';
+    my $ro      = '/srv/ro';
+    $stage //= '';
+    if ($stage eq 'before') {
+        install_service();
```
Comment (Vogtinator) [inline lib/nfs_common.pm]: Those three lines and the hardcoded paths are now duplicated across two files, can that be improved?

File: lib/nfs_common.pm (hunk 1 above)
Comment (coolgw) [reply lib/nfs_common.pm]: I just move the original code to here, not change anything. If you delete nfs-kernel-server, then you will encounter issue http://openqa.suse.de/tests/3264073#step/yast2_nfs_server/4 . 
So i would not change this.

File: lib/nfs_common.pm [hunk 6]
```diff
@@ -183,5 +186,74 @@ sub yast2_server_initial {
...
+    # Next step
+    send_key 'alt-n';
+
+    assert_screen 'nfs-overview';
+
+    add_shares($rw, $ro);
+
+    send_key 'alt-f';
+    wait_serial("$module_name-0") or die "'yast2 $module_name' didn't finish";
+
```
Comment (Vogtinator) [inline lib/nfs_common.pm]: Whitespace

Comment (Vogtinator) [inline lib/nfs_common.pm]: Whitespace

**Comment by coolgw** [comment]
@Vogtinator Could you help check again? I have made some update.

**Comment by Vogtinator** [comment]
There's still the same blank line, you should remove it a verification run with the latest state would be good for confirmation.

**Comment by coolgw** [comment]
@Vogtinator blank line deleted and update the today's result.

**Comment by lemon-suse** [comment]
LGTM.

**Pull request merged**